src/rust/engine/fs/src/directory.rs

```diff
@@ -311,6 +314,43 @@ impl From<&File> for remexec::FileNode {
   }
 }
 
+#[derive(Clone, Debug, DeepSizeOf)]
+pub struct Symlink {
+  name: Name,
+  target: String,
+}
+
+impl Symlink {
+  pub fn name(&self) -> Name {
+    self.name
+  }
+
+  pub fn target(&self) -> String {
+    self.target.to_owned()
```

Comment:
Generally you should expose a reference and let the caller decide whether they want to clone/to_owned it.
```suggestion
  pub fn target(&self) -> &str {
    &self.target
```